p2p.go

```diff
@@ -328,22 +331,27 @@ func NewHost(ctx context.Context, options ...Option) (*Host, error) {
 		}),
 		libp2p.Identity(prikey),
 		libp2p.DefaultSecurity,
-		libp2p.Transport(func(upgrader *tptu.Upgrader) *tcp.TcpTransport {
-			return &tcp.TcpTransport{Upgrader: upgrader, ConnectTimeout: cfg.ConnectTimeout}
+		libp2p.Transport(func(upgrader transport.Upgrader) *tcp.TcpTransport {
+			tp, err := tcp.NewTCPTransport(upgrader, &network.NullResourceManager{}, tcp.WithConnectionTimeout(cfg.ConnectTimeout))
+			if err != nil {
+				Logger().Panic("failed to new tcp transport", zap.Error(err))
+			}
+			return tp
 		}),
 		libp2p.Muxer("/yamux/2.0.0", yamux.DefaultTransport),
-		libp2p.ConnectionManager(connmgr.NewConnManager(cfg.ConnLowWater, cfg.ConnHighWater, cfg.ConnGracePeriod)),
+		libp2p.ConnectionManager(connMgr),
+		libp2p.DefaultPrometheusRegisterer,
 	}
 
 	if !cfg.SecureIO {
 		opts = append(opts, libp2p.NoSecurity)
 	} else {
-		opts = append(opts, libp2p.Security(secio.ID, secio.New))
+		// opts = append(opts, libp2p.Security(secio.ID, secio.New))
 	}
 
 	// relay option
 	if cfg.Relay == "active" {
-		opts = append(opts, libp2p.EnableRelay(relay.OptActive, relay.OptHop))
+		opts = append(opts, libp2p.EnableRelay())
```

Comment:
could you confirm

Reply:
there is no options any more in the new version.